### PR TITLE
[Post-Processors] Replaced ProcessBuilder with Process and add additional tests

### PR DESCRIPTION
**Pull request by fabianbloching**
| Q | A
| --- | ---
| Branch? | 2.0
| Bug fix? | yes
| New feature? | no
| BC breaks? | no
| Deprecations? | no
| Tests pass? | yes
| Fixed tickets | #1024 
| License | MIT

Pull request to support Symfony 4.0

### CODE REVIEW

**Comment by fabianbloching** [comment]
The phpunit 5.6 tests are not running, no tests are executed, see https://travis-ci.org/liip/LiipImagineBundle/jobs/316955919

**Comment by sebastianblum** [comment]
@dbu David, do you have an idea how to execute the unit tests with PHPUnit 5.7.
Also in other pull requests, not all tests are running, see https://travis-ci.org/liip/LiipImagineBundle/jobs/308329670

**Comment by dbu** [comment]
i played around in #1027 but still tests are not run. (i think the failure is because i try to cat phpunit.xml which does not exist)

when i run phpunit locally, tests are executed as expected. (i have to use simple-phpunit, otherwise i get the yaml error because phpunit loads a wrong version of the symfony yaml component)

**Comment by dbu** [comment]
can you please rebase this on the 2.0 branch to get the tests to run?

File: .travis.yml [hunk 1]
```diff
@@ -2,32 +2,41 @@ sudo: false
...
 matrix:
-  include:
-    - env: SYMFONY_VERSION=4.0.*
   fast_finish: true
-
+  include:
+    - php: 7.1
+      env: SYMFONY_VERSION=4.0.*
+    - php: 7.1
+      env: SYMFONY_VERSION=3.4.*
```
Comment (dbu) [inline .travis.yml]: i don't think we need to build with symfony 3.4 on more than 1 version of php. 7.2 is enough.

File: Imagine/Filter/PostProcessor/JpegOptimPostProcessor.php [hunk 2]
```diff
@@ -141,33 +141,33 @@ public function processWithConfiguration(BinaryInterface $binary, array $options
             throw new \RuntimeException(sprintf('Temp file can not be created in "%s".', $tempDir));
         }
 
-        $pb = new ProcessBuilder([$this->jpegoptimBin]);
+        $processArguments = [$this->jpegoptimBin];
 
         $stripAll = array_key_exists('strip_all', $options) ? $options['strip_all'] : $this->stripAll;
         if ($stripAll) {
-            $pb->add('--strip-all');
+            array_push($processArguments, '--strip-all');
```
Comment (dbu) [inline Imagine/Filter/PostProcessor/JpegOptimPostProcessor.php]: why not `$processArguments[] = '--strip-all';` ?

**Comment by dbu** [comment]
looking at the travis output, i also notice that some of the tests are skipped because of missing flysystem. i think we should have one build where we composer require those dependencies to know that the integration indeed works. (but not add them to composer.json require-dev, otherwise we don't see if other things accidentally depend on flysystem). but again, a task for a separate PR rather than mix it in here.

File: Imagine/Filter/PostProcessor/JpegOptimPostProcessor.php (hunk 2 above)
Comment (fabianbloching) [reply Imagine/Filter/PostProcessor/JpegOptimPostProcessor.php]: yeah, $processArguments[] = '--strip-all'; makes more sense. I refactored it.

**Comment by sebastianblum** [comment]
I created pull request #1035 to extend the build matrix

**Comment by fabianbloching** [comment]
@robfrawley I rebased against the master and now the merge conflicts are resolved. Maybe you have time to review this pr

**Comment by robfrawley** [comment]
@fabianbloching FYI, *merging* the upstream changes into your commit history creates a fragmented, redundant, and "un-clean" looking commit history. I've already fixed this PR, but in the future please try to **rebase** and not **merge** upstream changes. Here is a brief example (assuming your remotes are origin for your own repo and upstream for this repo):

```
git checkout -b 2.0-upstream upstream/2.0 # checkout upstream (or git pull if you already have it checked out)
git checkout my-feature-branch            # switch back to your feature branch
git rebase 2.0-upstream                   # rebase your feature branch on top of upstream changes
```

If you encounter any merge conflicts while your new commits are being replayed over the upstream changes, go ahead and manually fix the conflicts and then call `git add the-file-with-conflicts.ext` followed by `git rebase --continue`. If at any point you want to cancel the rebase issue `git rebase --abort`.

**Comment by robfrawley** [comment]
Thanks @fabianbloching!

**Comment by fabianbloching** [comment]
@robfrawley Thanks for the detailed briefing.

**Pull request merged**